### DISCUSSION TITLE
Custom Input Devices

### DISCUSSION THREAD

**Discussion by varneyc** [shanteacontrols/OpenDeck#57] [Q&A] [answered]
Hi all,

I've just gotten started on my own OpenDJ Build. Everything is going great so far, awesome documentation!

The one part of the build I am having issues with is my Jog Wheel setup. I am using custom-made jog wheels out of old hard drives, which I have got working very well to be read from an Arduino Pro Mini, but I am struggling to find a way to use the signals generated by the jog wheels on my OpenDeck setup (Arduino Mega2560). 

Some options I have considered are:
- Emulating a Rotary Encoder using the Arduino Pro Mini, and wiring that into the main board. Unfortunately, generating the signals is proving quite a challenge.
- Communicating between the Arduino Pro Mini and the main board over serial, although I do not see any way to take serial data as an input through OpenDeck
- Modifying the OpenDeck source code to process the jog wheel signal, but I do not see myself as quite skilled enough to manage this!

I'm seeing that some VERY old OpenDeck builds have made use of HDD platters as encoders, like in this article here from @paradajz : https://www.native-instruments.com/forum/threads/traktor-inspired-diy-midi-controller.179278/. 

Any ideas?

**Comment by paradajz** [comment, marked as answer]
That's actually my project you've linked. :) Maybe this old post I've written will be helpful

https://shanteacontrols.com/2015/02/24/building-ceylon/

**Reply by varneyc** [reply]
That is awesome! The schematics in that example closely resemble what I have been using. I really like the idea of the diodes to solve the -0.3V input issue, that is one of the main issues I have been facing.

My only concern is integration with the more recent OpenDeck versions, is there still support for this type of encoder, or will some modifications need to be made?

**Reply by varneyc** [reply]
Sorry, I should also mention I am using four-wire HDD motors, as opposed to the three-wire motors used in your post.

**Reply by paradajz** [reply]
I am not sure if the current OpenDeck firmware can handle that much information from spinning disk but you can try. Basically once you have A/B digital outputs, it's a simple matter of connecting it to two digital input pins and then enabling corresponding encoder in the interface.
